docs/topics/development/testing_and_quality_assurance.md

```diff
@@ -0,0 +1,152 @@
+
+- **Parallel Testing**:
+
+```sh
+pytest -n auto  # pytest-xdist
+```
+
+#### Configuration
+
+Configuration for your unit tests is handled automatically. Ensure that the database credentials in your settings have full permissions to modify a database with `test_` prepended to it. By default, the database name is `olympia`, so the test database is `test_olympia`.
+
+If the code you are working on is related to search, you'll want to run Elasticsearch tests. Ensure Elasticsearch is installed. See the :ref:`elasticsearch` page for details.
+
+- To exclude Elasticsearch tests:
+
+```sh
+make test_no_es
+```
+
+- To run only Elasticsearch tests:
+
+```sh
+make test_es
+```
+
+### Using `pytest` Directly
+
+For advanced users, you can connect to the Docker container using `make shell` and then use `pytest` directly, which allows for finer-grained control of the test suite.
+
+```sh
+pytest
+```
+
+Examples of running subsets of the test suite:
+
+- `pytest -m es_tests` to run tests marked as `es_tests`.
+- `pytest -k test_no_license` to run tests with `test_no_license` in their name.
+- `pytest src/olympia/addons/tests/test_views.py` to run all tests in a given file.
+- `pytest src/olympia/addons/tests/test_views.py::TestLicensePage` to run a specific test suite.
+- `pytest src/olympia/addons/tests/test_views.py::TestLicensePage::test_no_license` to run a specific test.
+
+For more details, see the [Pytest usage documentation](http://pytest.org/en/latest/usage.html#specifying-tests-selecting-tests).
+
+### Useful Makefile Targets
+
+- Run all tests:
+
+```sh
+make test
+```
+
+- Rebuild the database and run tests:
+
+```sh
+make test_force_db
+```
+
+- Stop on the first test failure:
+
+```sh
+make tdd
+```
+
+- Run tests with specific arguments or specific tests:
+
+```sh
+make test ARGS='-v src/olympia/amo/tests/test_url_prefix.py::MiddlewareTest::test_get_app'
+```
+
+- Re-run only the tests that failed in the previous run:
+
+```sh
+make test_failed
+```
+
+### Writing Tests
+
+We support two types of automated tests:
+
+- **Unit/Functional Tests**: Most tests fall into this category. Test classes extend `django.test.TestCase` and follow standard unit testing rules, using JSON fixtures for data.
+- **External Calls**: Avoid connecting to remote services in tests. Instead, mock out those calls with [responses](https://pypi.org/project/responses/).
+
+### Localization Tests
+
+If you want to test localization, add locales in the test directory (e.g., `devhub/tests/locale`). These locales should not appear unless added to `LOCALE_PATH`. If you change the `.po` files for these test locales, recompile the `.mo` files manually:
+
+```sh
+msgfmt --check-format -o django.mo django.po
+```
+
+## E2E Testing
+
+Our project includes end-to-end (E2E) tests written in a separate repository maintained by our QA team. Further documentation on E2E testing will be provided in the future.
```

Comment:
We could link the repos: https://github.com/mozilla/addons-release-tests. It contains instructions on how to run those tests.

Reply:
Done